skbio/io/registry.py

```diff
@@ -515,10 +515,15 @@ def _read_ret(self, file, fmt, into, verify, kwargs):
                 file.tell()
             except io.UnsupportedOperation:
                 seekable = False
-            if not seekable:
-                print(seekable)
-            if not seekable and (fmt is not None or not fmt.support_non_seekable):
-                raise ValueError("Cannot parse non-seekable data of type %r", fmt.name)
+            if not seekable and fmt is None:
```

Comment:
This works but `not seekable` is checked twice. It can be combined into one check.

```python
if not seekable:
    if fmt is None:
        raise ValueError("Cannot sniff filetype of non-seekable data.")
    for lookup in self._lookups:
        ...
```